typescript/internal/src/AuthForm.tsx

```diff
@@ -0,0 +1,49 @@
+import type { AuthManager } from "./authManager";
+import { useForm } from "react-hook-form";
+import { useMutation } from "@tanstack/react-query";
+import { useId } from "react";
+
+interface AuthFormData {
+  handle?: string;
+}
+
+interface AuthFormProps {
+  authManager: AuthManager;
+  redirectUrl: string;
+}
+
+export default function AuthForm({ authManager, redirectUrl }: AuthFormProps) {
+  const { register, handleSubmit } = useForm<AuthFormData>();
+  const {
+    mutate: login,
+    isPending,
+    error,
+    isError,
+  } = useMutation({
+    async mutationFn({ handle }: AuthFormData) {
+      if (!handle) {
+        throw new Error("Handle required");
+      }
+      const url = authManager.loginUrl(handle, redirectUrl);
+      window.location.href = url.toString();
```

Comment:
Is there a reason to use this rather than the routing framework to change the page location?
 Might be a bit tough because that could vary based on what is importing this component.

Reply:
the routing framework is irrelevant here. We need to do an external page nav which doesn't depend on the routing framework at all.